CONTRIBUTING.md

```diff
@@ -302,92 +338,183 @@ into any of those categories, use your best judgement.
+- `tests/pytype_test.py` tests typeshed with
+[pytype](https://github.com/google/pytype/).
+- `tests/pyright_test.py` tests typeshed with
+[pyright](https://github.com/microsoft/pyright).
+- `tests/mypy_test_suite.py` runs a subset of mypy's test suite using this version of
+typeshed.
+- `tests/check_consistent.py` checks certain files in typeshed remain
+consistent with each other.
+- `tests/stubtest_test.py` checks stubs against the objects at runtime.
+- `flake8` enforces a style guide.
+
+### Setup
+
+Run:
+```
+$ python3 -m venv .venv3
+$ source .venv3/bin/activate
+(.venv3)$ pip install -U pip
+(.venv3)$ pip install -r requirements-tests-py3.txt
+```
+This will install mypy (you need the latest master branch from GitHub),
+typed-ast, flake8 (and plugins), pytype, black and isort.
 
-### Stub versioning
+If you want to run the pyright tests, you need to have
+[Node.js](https://nodejs.org/) installed.
 
-There are separate directories for `stdlib` (standard library) and `stubs`
-(all other stubs). For standard library stubs Python version support is
-given in `VERSIONS` file. Each line in this file is a module or package name
-followed by `: `, followed by the oldest *supported* Python version where
-the module is available.
+### mypy\_test.py
 
-Third-party stubs only support Python 3 by default. You can optionally supply
-Python 2 stubs for a package by placing them into a `@python2` subdirectory
-for the corresponding distribution. Some older stubs also indicate Python 2
-support by setting `python2 = True` in the corresponding `METADATA.toml` file.
+This test requires Python 3.6 or higher; Python 3.6.1 or higher is recommended.
+Run using:
+```
+(.venv3)$ python3 tests/mypy_test.py
+```
 
-You can use checks
-like `if sys.version_info >= (3, 8):` to denote new functionality introduced
-in a given Python version or solve type differences.  When doing so, only use
-one-tuples or two-tuples.  This is because:
+This test is shallow — it verifies that all stubs can be
+imported but doesn't check whether stubs match their implementation
+(in the Python standard library or a third-party package). It has an exclude list of
+modules that are not tested at all, which also lives in the tests directory.
 
-* mypy doesn't support more fine-grained version checks; and more
-  importantly
+You can restrict mypy tests to a single version by passing `-p2` or `-p3.9`:
+```bash
+(.venv3)$ python3 tests/mypy_test.py -p3.9
+```
 
-* the micro versions of a Python release will change over time in your
-  checking environment and the checker should return consistent results
-  regardless of the micro version used.
+### pytype\_test.py
 
-Because of this, if a given functionality was introduced in, say, Python
-3.7.4, your check:
+This test requires Python 2.7 and Python 3.6. Pytype will
+find these automatically if they're in `PATH`, but otherwise you must point to
+them with the `--python27-exe` and `--python36-exe` arguments, respectively.
```

Comment:
I think these args / versions are out of date

Reply:
Removing.